application/apps/indexer/processor/src/text_source.rs

```diff
@@ -308,3 +347,41 @@ fn count_lines_up_to_last_newline(buffer: &[u8]) -> (u64, usize) {
         (0, 0)
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::bytes_to_utf8;
+    use proptest::prelude::*;
+    use rand::{thread_rng, Rng};
+
+    fn strategy() -> impl Strategy<Value = Vec<u8>> {
```

Comment:
better then to create a strategy from hand, use the proptest way to compose:
``` rust
    prop_compose! {
        fn sample_string()
                    (mut pre in prop::collection::vec(any::<u8>(), 0..=0x7F),
                     mut post in prop::collection::vec(any::<u8>(), 0x80..=0xBF))
                    -> Vec<u8> {
                        let mut res: Vec<u8> = vec![];
                        res.append(&mut pre);
                        res.push(0xC1);
                        res.append(&mut post);
                        res
        }
    }
```

Reply:
Thanks for it. Have no idea how I've missed it... with it indeed the fix became trivial

Reply:
No need to test (at least on this level) as soon as there are no manual implementation any more.